tests/test_ext_autosummary.py

```diff
@@ -254,6 +255,64 @@ def test_autosummary_generate_overwrite2(app_params, make_app):
     assert 'autosummary_dummy_module.rst' not in app._warning.getvalue()
 
 
+@pytest.mark.sphinx('dummy', testroot='ext-autosummary-recursive',
+                    srcdir='ext-autosummary-recursive-enabled',
+                    confoverrides={'autosummary_recursive': True})
+def test_autosummary_recursive_enabled(app, status, warning):
+    app.build()
+    toctree = 'modules'  # see module.rst template
+
+    # Top-level package
+    generated = app.srcdir / 'generated'
+    assert (generated / 'package.rst').exists()
+    content = (generated / 'package.rst').text()
+    assert 'package.module' in content
+    assert 'package.package' in content
+
+    # Recursively generate modules of top-level package
+    generated /= toctree
+    assert (generated / 'package.module.rst').exists()
+    assert (generated / 'package.package.rst').exists()
+    content = (generated / 'package.package.rst').text()
+    assert 'package.package.module' in content
+    assert 'package.package.package' in content
+
+    # Recursively generate modules of sub-package
+    generated /= toctree
+    assert (generated / 'package.package.module.rst').exists()
+    assert (generated / 'package.package.package.rst').exists()
+    content = (generated / 'package.package.package.rst').text()
+    assert 'package.package.package.module' in content
+    assert 'package.package.package.package' not in content
```

Comment:
These deeper packages might be needed for checking the behavior of recursive_limit. But they're no longer needed now.